app/components/computed-props.js

```diff
@@ -50,11 +51,88 @@ export default Component.extend({
    * Please replace these props below with computed props
    * @type {Array}
    */
-  userNames: [],
-  bmwusers: [],
-  whiteCarsUsers: [],
-  usersWith1Car: [],
-  userswithMoreThanOneCar: [],
-  allMakes: [] // no repeat of makes
+  userNames: computed('users', function () {
+    const userNames = [];
+    this.get('users').forEach((item) => {
+      userNames.push(item.name);
+    });
+    return userNames;
+  }),
+  bmwusers: computed('users', function () {
+    const bmwUsers = [];
+    this.get('users').forEach((user) => {
```

Comment:
supposing you have a `user` model (if you don't forget it), then you would first declare a `foundCar` computed property (CP) in the `user` model

Reply:
and then a `computed.filterBy('users', 'foundCar', true)`

Reply:
wouldn't this lead to cluttering the user model with lots of computed properties? as foundCar is only used in one specific use case.

Reply:
good question. Yes if you only use this CP once. No if you actually start reusing all those properties in every app.

Reply:
and we can (will) talk about it for hours, but in a MVC controller, models are naturally created to contain such business logic, filtering etc.

Reply:
I _think_ it is the same debate as https://www.toptal.com/software/declarative-programming ?